internal/collector/job.go

```diff
@@ -255,6 +257,44 @@ var (
 					})
 				}
 
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_job_owner",
+			Type: metric.MetricTypeGauge,
+			Help: "Information about the Job's owner.",
+			GenerateFunc: wrapJobFunc(func(j *v1batch.Job) *metric.Family {
+				labelKeys := []string{"owner_kind", "owner_name", "owner_is_controller"}
+				ms := []*metric.Metric{}
+
+				owners := j.GetOwnerReferences()
+				if len(owners) == 0 {
+					ms = append(ms, &metric.Metric{
+						LabelKeys:   labelKeys,
+						LabelValues: []string{"<none>", "<none>", "<none>"},
```

Comment:
no metric should be appended if none exists, in prometheus that's queryable through the `absent` function

Reply:
Aye, I've had the same concern, but I've taken the Pod collector as a reference. There is no such machinery there.
https://github.com/kubernetes/kube-state-metrics/blob/master/internal/collector/pod.go#L121

Reply:
Might be good to leave at it is to preserve uniformity in how we present metrics.

Reply:
Good point. @brancz as changing this in `pod.go` would be a breaking change, should we do this with the v2.0 release?

Reply:
Let's do it right in this case, and add it to the 2.0 release tracking issue: #569.

Reply:
@zuzzas IIUC, you need to delete the logic for appending `<none>` value metrics when the owners are empty as what @brancz proposed in comment https://github.com/kubernetes/kube-state-metrics/pull/681/files#r259773399

Reply:
That I understood, @andyxning. My question was about the case in Pod collector, which works differently. Right now my issues are resolved any I'm adding a new feature.

Reply:
I've done that in a separate commit: https://github.com/kubernetes/kube-state-metrics/pull/681/commits/894b9af069cb1dddd6a93d3bc6bd3622ffdbbe21